### PR TITLE
TrueTypeFont: Check for enough data to unpack

### PR DESCRIPTION
**Pull request by sschuberth**
Fixes https://github.com/euske/pdfminer/issues/96.

### CODE REVIEW

**Comment by pombredanne** [comment]
@sschuberth Thanks! This might work though I am not enough of a PDF expert to make the changes are sensible... and this would need a unit test of sorts.
@goulu there is a PDF test file here: https://github.com/nexB/scancode-toolkit/tree/6d07756efcef9b6014f553dd1f084a16cfc1f474/tests/scancode/data/failing 
Do you want an enhanced patch with a test?

**Comment by pombredanne** [comment]
It works with my test PDF.

**Comment by sschuberth** [comment]
Let me try to add a test with your myself.

File: pdfminer/pdffont.py [hunk 1]
```diff
@@ -383,9 +383,15 @@ def __init__(self, name, fp):
         self.fp = fp
         self.tables = {}
         self.fonttype = fp.read(4)
-        (ntables, _1, _2, _3) = struct.unpack('>HHHH', fp.read(8))
+        data_str = fp.read(8)
+        if (len(data_str) != 8):
```
Comment (pombredanne) [inline pdfminer/pdffont.py]: There is no need for the extra () eg. `if len(data_str) != 8:` is enough

File: pdfminer/pdffont.py [hunk 2]
```diff
@@ -383,9 +383,15 @@ def __init__(self, name, fp):
...
+            return
+        (ntables, _1, _2, _3) = struct.unpack('>HHHH', data_str)
         for _ in range(ntables):
-            (name, tsum, offset, length) = struct.unpack('>4sLLL', fp.read(16))
+            data_str = fp.read(16)
+            if (len(data_str) != 16):
```
Comment (pombredanne) [inline pdfminer/pdffont.py]: Same as above: there is no need for the extra () eg. `if len(data_str) != 8:` is enough

File: tests/test_tools_pdf2txt.py [hunk 3]
```diff
@@ -47,11 +47,15 @@ def test_6(self):
...
- 
+
     def test_8(self):
         run('../samples/contrib/','2b','-A -t xml')
 
     def test_9(self):
         run('../samples/nonfree/','175') # https://github.com/pdfminer/pdfminer.six/issues/65
+
+    def test_10(self):
+        run('../samples/scancode/','patchelf.pdf') # https://github.com/euske/pdfminer/issues/96
```
Comment (pombredanne) [inline tests/test_tools_pdf2txt.py]: Sounds like the test environment adds a `.pdf` extension for you so this should be `run('../samples/scancode/','patchelf')` instead

**Comment by pombredanne** [comment]
NB: this likely also fixes: https://github.com/euske/pdfminer/issues/144

**Comment by pombredanne** [comment]
@goulu @vstoykov any feedback? It looks good to me to merge .... and this is triggering crashes in scancode ;)

**Comment by vstoykov** [comment]
From the Python glossary

> Easier to ask for forgiveness than permission.

I think that is better to surround the unpack with `try:..except:` block catching the right exception instead of storing the data in variable and checking it's length.

**Comment by sschuberth** [comment]
@vstoykov, done!

**Comment by sschuberth** [comment]
Any update regarding my changes that use `try:..except:`, @vstoykov?

**Comment by vstoykov** [comment]
I'm ok with the change. Probably some comment why in the except block we only `pass`.

The final word will be from @goulu.

**Comment by sschuberth** [comment]
> Probably some comment why in the except block we only `pass`.

Done.

**Comment by pombredanne** [comment]
@goulu it would be awesome if you could find a few minutes to review this and push a Pypi release
Thank you in advance!

**Comment by henryl** [comment]
Can we merge this? Has been sitting here for a while and we are experiencing the same bug.

**Comment by pombredanne** [comment]
@goulu is there something we can do to help you with merging this?

**Comment by ganeshtata** [comment]
I am not sure if @goulu is available for merging this request, but if @pombredanne is confident enough for this merge, I can do it, although I don't possess the required knowledge for reviewing this pull request.

**Comment by pombredanne** [comment]
@ganeshtata Thanks. Yes I am confident with this merge that I reviewed and tested.

Signed-off-by: Philippe Ombredanne <pombredanne@nexb.com>

**Comment by ganeshtata** [comment]
@pombredanne  I will go with your assurance, and merge the request.

**Comment by pombredanne** [comment]
@ganeshtata Thanks!

**Comment by pombredanne** [comment]
@ganeshtata could you also tag and push to Pypi?

**Comment by ganeshtata** [comment]
@pombredanne Sure. But, if possible, could you please guide me as to how should I push to Pypi?

**Comment by pombredanne** [comment]
@ganeshtata `twine upload ...` but do you have the creds to the Pypi account?

**Comment by ganeshtata** [comment]
@pombredanne Yes, I do have creds to the Pypi account.

**Pull request merged**